src/model/building.rs

```diff
@@ -1,13 +1,67 @@
+use rocket::http::RawStr;
+use rocket::request::FromFormValue;
+
 /**
  * A building housing floors
  */
 #[derive(Serialize, Deserialize)]
 pub struct Building {
-    id: String,
-    name: String,
-    address: String,
-    phone_number: String,
-    email: String,
-    geo_coordinates: (f32, f32),
-    model_type: String,
-}
\ No newline at end of file
+    pub id: String,
+    pub name: String,
+    pub address: String,
+    pub phone_number: String,
+    pub email: String,
+    pub geo_coordinate: GeoCoordinate,
+    pub model_type: String,
+}
+
+#[derive(FromForm, Serialize, Deserialize)]
+pub struct GeoCoordinate {
+    pub lat: f32,
+    pub lng: f32,
+}
+
+impl<'v> FromFormValue<'v> for GeoCoordinate {
+    type Error = &'v RawStr;
+
+    fn from_form_value(form_value: &'v RawStr) -> Result<GeoCoordinate, &'v RawStr> {
```

Comment:
Might be a good candidate for unit testing

Reply:
Added 4 test cases for it